ddtrace/sampler.py

```diff
@@ -235,10 +235,8 @@ def __init__(
         super(DatadogSampler, self).__init__()
 
         if default_sample_rate is None:
-            sample_rate = ddconfig._trace_sample_rate
-
-            if sample_rate is not None:
-                default_sample_rate = float(sample_rate)
```

Comment:
This was a case where we had some funky logic in place to try to infer if the setting was user-defined or not.